alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteMonitorAction.kt

```diff
@@ -74,6 +91,65 @@ class TransportExecuteMonitorAction @Inject constructor(
                     }
                 }
             }
+            val indexDocLevelMonitorQueries = fun(monitor: Monitor, monitorId: String) {
```

Comment:
Do we really need to update the mapping each time during the monitor run execution? Shouldn't the Index Monitor path take care of any update in the mapping?

Reply:
there are 2 types of execute action. one which takes `already created monitor_id` as input. the other one directly takes entire monitor payload as input.
update mapping is needed in 2nd case.
https://github.com/opensearch-project/alerting/blob/79a368caa73b64ab08b0cd14c4da0324ded8c1db/alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteMonitorAction.kt#L89-#L180